case/src/main/kotlin/com/ritense/case/web/rest/dto/CaseDefinitionDraftCreateRequest.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.case.web.rest.dto
+
+import com.ritense.valtimo.contract.case_.CaseDefinitionId
+
+data class CaseDefinitionDraftCreateRequest(
+    val caseDefinitionKey: String,
+    val caseDefinitionVersion: String,
+    val name: String? = null,
+    val description: String? = null,
+    val basedOnCaseDefinitionVersion: String? = null,
+) {
+    fun getCaseDefinitionId(): CaseDefinitionId = CaseDefinitionId(caseDefinitionKey, caseDefinitionVersion)
+
+    fun getBasedOnCaseDefinitionId(): CaseDefinitionId? {
```

Comment:
Since we still have time to change this: Should we rename this to `getReferenceCaseDefinitionId`? Or `getOriginalCaseDefinitionId`?

Reply:
This is Git Branch terminology. I the future it might be possible to 'rebase' the case-definition on another case-definition.